src/app/auth/custom-input-field/custom-input-field.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Component, Input } from '@angular/core';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+
+@Component({
+  selector: 'app-custom-input-field',
+  standalone: true,
+  imports: [MatFormFieldModule, ReactiveFormsModule, MatInputModule],
+  templateUrl: './custom-input-field.component.html',
+  styleUrl: './custom-input-field.component.scss',
+})
+export class CustomInputFieldComponent {
+  @Input({required: true}) public label!: string;
+  @Input({required: true}) public placeholder!: string;
+  @Input() public type = 'text';
```

Comment:
You do not need an email type since that will be control by `FormControl`
```suggestion
  @Input() public type: 'text' | 'password' = 'text';
```